plugin/storage/kafka/options_test.go

```diff
@@ -45,4 +53,144 @@ func TestFlagDefaults(t *testing.T) {
 	assert.Equal(t, defaultTopic, opts.topic)
 	assert.Equal(t, []string{defaultBroker}, opts.config.Brokers)
 	assert.Equal(t, defaultEncoding, opts.encoding)
+	assert.Equal(t, sarama.WaitForLocal, opts.config.RequiredAcks)
+	assert.Equal(t, sarama.CompressionNone, opts.config.Compression)
+	assert.Equal(t, 0, opts.config.CompressionLevel)
+}
+
+func TestCompressionLevelDefaults(t *testing.T) {
+	compressionLevel, err := getCompressionLevel("none", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["none"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("gzip", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["gzip"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("snappy", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["snappy"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("lz4", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["lz4"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("zstd", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["zstd"].defaultCompressionLevel, compressionLevel)
+}
+
+func TestGetCompressionLevel(t *testing.T) {
+	compressionLevel, err := getCompressionLevel("none", 0)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["none"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("gzip", 4)
+	require.NoError(t, err)
+	assert.Equal(t, 4, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("snappy", 0)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["snappy"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("lz4", 10)
+	require.NoError(t, err)
+	assert.Equal(t, 10, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("zstd", 20)
+	require.NoError(t, err)
+	assert.Equal(t, 20, compressionLevel)
+}
+
+func TestFailedCompressionLevelScenario(t *testing.T) {
+	_, err := getCompressionLevel("gzip", 14)
+	require.Error(t, err)
+
+	_, err = getCompressionLevel("lz4", 18)
+	require.Error(t, err)
+
+	_, err = getCompressionLevel("zstd", 25)
+	require.Error(t, err)
+}
+
+func TestGetCompressionModes(t *testing.T) {
+	compressionModes, err := getCompressionModes("gzip")
```

Comment:
s/compressionModes/compression/

Reply:
fixed

Reply:
fixed?  I'm still seeing `compressionModes` variable, whereas it's a single `compression` value